service/header/core_listener.go

```diff
@@ -40,7 +37,7 @@ func NewCoreListener(p2pSub *P2PSubscriber, fetcher *core.BlockFetcher, dag form
 
 // Start kicks off the CoreListener listener loop.
 func (cl *CoreListener) Start(ctx context.Context) error {
-	if cl.blockSub != nil {
+	if cl.cancel != nil {
```

Comment:
We should always use the cancel func as a way to check the lifecycle state. It won't ever cause us races.